cw_bitcoin/lib/litecoin_wallet_addresses.dart

```diff
@@ -15,15 +30,101 @@ abstract class LitecoinWalletAddressesBase extends ElectrumWalletAddresses with
     required super.mainHd,
     required super.sideHd,
     required super.network,
+    required this.mwebHd,
+    required this.mwebEnabled,
     super.initialAddresses,
     super.initialRegularAddressIndex,
     super.initialChangeAddressIndex,
-  }) : super(walletInfo);
+  }) : super(walletInfo) {
+    if (mwebEnabled) {
+      // give the server a few seconds to start up before trying to get the addresses:
+      Future.delayed(const Duration(seconds: 5), () async {
+        await topUpMweb(0);
+      });
+    }
+  }
+
+  final Bip32Slip10Secp256k1 mwebHd;
+  bool mwebEnabled;
+
+  List<int> get scanSecret => mwebHd.childKey(Bip32KeyIndex(0x80000000)).privateKey.privKey.raw;
+  List<int> get spendPubkey =>
+      mwebHd.childKey(Bip32KeyIndex(0x80000001)).publicKey.pubKey.compressed;
+
+  List<String> mwebAddrs = [];
+
+  Future<void> topUpMweb(int index) async {
+    // generate up to index + 1000 addresses:
+    while (mwebAddrs.length - index < 1000) {
```

Comment:
why 1000?

Reply:
this was the original behavior, but it's a good question, I'll ask hector just to double check if it's really necessary